model_signing/signature/signing.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This package provides the functionality to sign models"""
+
+from typing import Optional
+
+import abc
+
+from absl import logging as log
+from cryptography.hazmat.primitives import serialization as crypto_serialization
+from cryptography.hazmat.primitives.asymmetric import ec as crypto_ec
+from cryptography.hazmat.primitives.hashes import SHA256
+from cryptography import x509 as crypto_x509
+from google.protobuf import json_format
+from in_toto_attestation.v1 import statement
+from sigstore import dsse
+from sigstore import oidc
+from sigstore import sign
+from sigstore_protobuf_specs.dev.sigstore.bundle import v1 as bundle_pb
+from sigstore_protobuf_specs.dev.sigstore.common import v1 as common_pb
+from sigstore_protobuf_specs.io import intoto as intoto_pb
+
+from signature import utils
+
+
+class Signer(abc.ABC):
```

Comment:
Can we separate abstract classes to main file (this one) an and implementation to their own files? Ideally we'd allow people to only import the files that contain the implementation they need so imports are fast and not all dependencies are needed

Reply:
I split out the implementations into files containing both the signer and the verifier.